app/storage/postgres/tenant.go

```diff
@@ -205,6 +205,14 @@ func (s *TenantStorage) UpdateSettings(settings *models.UpdateTenantSettings) er
 			newLogoID.Scan(uploadID)
 		}
 
+		// Try to find out if there is an upload for the logo id of the tenant
+		oldLogoExists, err := s.trx.Exists("SELECT content_type, size, file FROM uploads WHERE id = $1", s.current.LogoID)
```

Comment:
```suggestion
		oldLogoExists, err := s.trx.Exists("SELECT id FROM uploads WHERE id = $1 AND tenant_id = $2", s.current.LogoID, s.current.ID)
```

If we don't need all those fields, better return just one. We also need to make sure tenant_id is always filtered on all SQL commands.